WordPress/gutenberg
# Consolidate template actions components

## What?
This PR refactors the code to use the same `TemplateActions` component, that is used in templates and template parts `more` menu(ellipsis). Currently there is a mismatch in provided actions between the sidebar navigation and the template/template part cart menu.


## Notes
In trunk `TemplateActions` are also used for patterns, which is not ideal since it works for template parts but not for the rest of patterns. We need to provide the available actions there too for patterns.

## Testing Instructions
1. Navigate through templates and template parts and observe that the same actions are provided in both places and work as expected.

## Screenshots or screencast 


https://github.com/WordPress/gutenberg/assets/16275880/8e28d59b-2fd6-4f3a-94cc-e9f4cdd23f9d
